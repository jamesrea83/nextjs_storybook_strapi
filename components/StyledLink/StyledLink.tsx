import Link from 'next/link';
import styled from '@emotion/styled';

export const StyledLink = styled(Link)`
	all: unset;
	cursor: pointer;
	color: ${({ theme }) => theme.font.regular};
	&:hover {
		opacity: 0.7;
	}
`;
