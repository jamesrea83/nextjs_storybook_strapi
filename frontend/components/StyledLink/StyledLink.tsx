import Link from 'next/link';
import styled from '@emotion/styled';

export type Props = {
	underline?: number;
};

export const StyledLink = styled(Link)<Props>`
	all: unset;
	cursor: pointer;
	color: ${({ theme }) => theme.font.regular};
	text-decoration: ${({ underline }) => (underline ? 'underline' : 'none')};
	&:hover {
		opacity: 0.7;
	}
`;
