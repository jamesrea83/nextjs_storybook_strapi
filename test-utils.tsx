import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from './styles/themes';

interface Props {
	children: ReactNode;
}

const Wrapper = ({ children }: Props) => (
	<ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
	render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
