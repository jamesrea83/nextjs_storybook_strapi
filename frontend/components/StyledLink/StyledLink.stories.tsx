import { Meta, StoryObj } from '@storybook/react';
import { expect } from '@storybook/jest';
import { screen } from '@storybook/testing-library';

import { StyledLink } from './StyledLink';

export default {
	title: 'Content/StyledLink',
	component: StyledLink,
} as Meta<typeof StyledLink>;

type Story = StoryObj<typeof StyledLink>;

export const Primary: Story = {
	play: async ({}) => {
		await expect(screen.getByRole('link')).toBeInTheDocument();
	},
	args: {
		children: 'Styled Link',
		href: 'test-link',
	},
};
