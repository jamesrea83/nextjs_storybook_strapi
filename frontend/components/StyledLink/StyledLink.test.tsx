import { render } from '@/test-utils';

import { StyledLink } from './StyledLink';

describe('StyledLink tests', () => {
	it('should render', () => {
		const { asFragment } = render(
			<StyledLink href="test-link">Link Test</StyledLink>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
