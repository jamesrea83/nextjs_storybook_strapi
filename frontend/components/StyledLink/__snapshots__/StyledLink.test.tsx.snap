// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledLink tests should render 1`] = `
<DocumentFragment>
  <a
    class="css-nzbn0p"
    href="test-link"
  >
    Link Test
  </a>
</DocumentFragment>
`;
